graphql/resolvers.go

```diff
@@ -3963,6 +3966,58 @@ func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion)
 	return status, nil
 }
 
+func (*versionResolver) UpstreamProject(ctx context.Context, obj *restModel.APIVersion) (*UpstreamProject, error) {
+	v, err := model.VersionFindOneId(utility.FromStringPtr(obj.Id))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding version %s: %s", *obj.Id, err.Error()))
+	}
+	if v == nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Version %s not found", *obj.Id))
+	}
+	if v.TriggerID == "" || v.TriggerType == "" {
+		return nil, nil
+	}
+	var projectID string
+	var revision string
+	if v.TriggerType == model.ProjectTriggerLevelTask {
+		upstreamTask, err := task.FindOneId(v.TriggerID)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding upstream task %s: %s", v.TriggerID, err.Error()))
+		}
+		if upstreamTask == nil {
+			return nil, errors.New("upstream task not found")
+		}
+		projectID = upstreamTask.Project
+		revision = upstreamTask.Revision
+	} else if v.TriggerType == model.ProjectTriggerLevelBuild {
+		upstreamBuild, err := build.FindOneId(v.TriggerID)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding upstream build %s: %s", v.TriggerID, err.Error()))
+		}
+		if upstreamBuild == nil {
+			return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Upstream build %s not found", v.TriggerID))
+		}
+		projectID = upstreamBuild.Project
+		revision = upstreamBuild.Revision
+	}
+	upstreamProject, err := model.FindBranchProjectRef(projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding upstream project")
```

Comment:
I think to be in line with the updates Kim is making for error styling, these should omit the word error going forward (i.e. just "finding upstream project"), although I'm not sure if she's intending to also make these changes to the graphql package.